app/api/server/v1/voip/omnichannel.ts

```diff
@@ -0,0 +1,94 @@
+import { Match, check } from 'meteor/check';
+
+import { API } from '../../api';
+import { Users } from '../../../../models/server/index';
+import { hasPermission } from '../../../../authorization/server/index';
+import { LivechatVoip } from '../../../../../server/sdk';
+
+API.v1.addRoute('omnichannel.agent.extension', { authRequired: true }, {
+	// Get the extensions associated with the agent passed as request params.
+	get() {
+		if (!hasPermission(this.userId, 'query-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.requestParams(), Match.ObjectIncluding({
+			username: String,
+		}));
+		const user = Users.findOneByAgentUsername(this.requestParams().username, {
+			fields: { _id: 1 },
+		});
+		if (!user) {
+			return API.v1.notFound();
+		}
+		const extension = Users.getExtension(user._id, {
+			fields: { _id: 1, username: 1, extension: 1 },
+		});
+		if (!extension) {
+			return API.v1.notFound();
+		}
+		return API.v1.success(...extension);
+	},
+
+	// Create agent-extension association.
+	post() {
+		if (!hasPermission(this.userId, 'managage-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.bodyParams, {
+			username: String,
+			extension: String,
+		});
+		const user = Users.findOneByAgentUsername(this.bodyParams.username, {
+			fields: { _id: 1, username: 1 },
+		});
+		if (!user) {
+			return API.v1.notFound();
+		}
+		Users.setExtension(user._id, this.bodyParams.extension);
+		return API.v1.success();
+	},
+	delete() {
+		if (!hasPermission(this.userId, 'managage-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.requestParams(), Match.ObjectIncluding({
+			username: String,
+		}));
+		const user = Users.findOneByAgentUsername(this.requestParams().username, {
+			fields: { _id: 1, username: 1 },
+		});
+		if (!user) {
+			return API.v1.notFound();
+		}
+		Users.unsetExtension(user._id);
+		return API.v1.success();
+	},
+});
+
+// Get free extensions
+API.v1.addRoute('omnichannel.extension.free', { authRequired: true }, {
+	get() {
+		if (!hasPermission(this.userId, 'managage-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		const extension = Promise.await(LivechatVoip.getAvailableExtensions());
```

Comment:
```suggestion
		const extension = await LivechatVoip.getAvailableExtensions();
```

Reply:
Since its already a typescript file, we can use this shortcut instead of calling Promise.await.

Reply:
This does not seem to work as expected. If we simply want to use async, we need to make the get function async.
But framework doesnt seem to call get in await mode. So whatever is returned is some object which does not have the desired result. I tried to see if I can find the type. But that too is not present. So we have to use Promise.await, which will not need the get method to be async.

Reply:
Okay, that makes sense. Please ignore this suggestion